notifier/notifier.go

```diff
@@ -712,6 +712,16 @@ func (s *alertmanagerSet) sync(tgs []*targetgroup.Group) {
 		seen[us] = struct{}{}
 		s.ams = append(s.ams, am)
 	}
+	// Now remove counters for any removed Alertmanagers
+	for _, am := range allAms {
+		us := am.url().String()
+		if _, ok := seen[us]; !ok {
+			continue
+		}
+		s.metrics.sent.DeleteLabelValues(us)
+		s.metrics.errors.DeleteLabelValues(us)
+		seen[us] = struct{}{}
+	}
```

Comment:
I think the logic is off here.

IIUC this would go through all AMs and delete the metrics for all those that we just have added.

How about this:

```suggestion
	for _, am := range allDroppedAms {
		us := am.url().String()
		if _, ok := seen[us]; ok {
			// Do not remove Alertmanagers that we have just added.
			continue
		}
		s.metrics.sent.DeleteLabelValues(us)
		s.metrics.errors.DeleteLabelValues(us)
	}
```

Note that `DeleteLabelValues` is fine with no-ops, i.e. an attempt to delete a metric that doesn't exist.